src/adapter/src/coord.rs

```diff
@@ -2941,6 +2952,43 @@ impl Coordinator {
             let idle_metric = self.metrics.queue_busy_seconds.with_label_values(&[]);
             let last_message_watchdog = Arc::clone(&last_message);
 
+            // TODO(jkosh44) Consider doing this periodically instead of only on startup.
+            if let Some(retention_period) = storage_usage_retention_period {
+                let id = self.catalog().resolve_builtin_table(&MZ_STORAGE_USAGE_BY_SHARD);
+                let read_ts = self.get_local_read_ts().await;
+                let current_contents_fut = self
+                    .controller
+                    .storage
+                    .snapshot_latest_async(id)
+                    .await
+                    .unwrap_or_terminate("cannot fail to fetch snapshot");
```

Comment:
I'm seeing a panic here for an unknown ID, maybe I should be using `self.controller.storage_collections` instead.